### PR TITLE
[CORE][SQL][MINOR] Scaladoc fixes + string interpolation

### PR DESCRIPTION
**Pull request by jaceklaskowski**
## What changes were proposed in this pull request?

Scaladoc fixes + string interpolation for logging
## How was this patch tested?

local build + manual testing

### CODE REVIEW

**Comment by srowen** [comment]
Some of this is OK, some of it isn't an improvement. I don't think we should just change a few instances of format to interpolation, nor do I think it is necessarily worth it to change everything all at once. Same with stuff like `nonEmpty`. This is just too hodge-podge to bother with.

File: yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnScheduler.scala [hunk 1]
```diff
@@ -31,9 +31,8 @@ private[spark] class YarnScheduler(sc: SparkContext) extends TaskSchedulerImpl(s
     Logger.getLogger(classOf[RackResolver]).setLevel(Level.WARN)
   }
 
-  // By default, rack is unknown
   override def getRackForHost(hostPort: String): Option[String] = {
-    val host = Utils.parseHostPort(hostPort)._1
+    val (host, _) = Utils.parseHostPort(hostPort)
```
Comment (rxin) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnScheduler.scala]: is this even an improvement?

Comment (jaceklaskowski) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnScheduler.scala]: It is when you agree that you could easily (?) miss `_1` at the very end. I do agree and I did miss it few times while reviewing that piece of code. Opinions may vary and I can happily revert this change if requested (I need your advice to learn your coding style).

**Comment by jaceklaskowski** [comment]
The issue with `nonEmpty` is that you could easily miss the negation (and that's why Scala offers `nonEmpty`). I don't think it's the final solution, but certainly believe it's far better for future readers to know what's going on in these lines. I believe the changes improve readability (but can happily revert _some_ if you point me at the places that need this).

It's just me to believe that by doing these small changes the code becomes more readable. I spent enough time with it to think it needs so (more often than I'm proposing).

Please guide me to learn your coding style. Thanks!

**Comment by rxin** [comment]
I'm with @srowen on this one. Not sure about the "improvements" here are indeed improvements.

**Comment by jaceklaskowski** [comment]
How am I supposed to read this? Do you want me to...forget about the changes? All of them or just some? Which one would you accept since @srowen said: "Some of this is OK"? It _appears_ there's _some_ merit in the change.

**Comment by rxin** [comment]
In this cases it seems like the value is pretty small. I have another pr that does some api annotation cleanups. I will port your typo fixes over to that.

**Comment by srowen** [comment]
In roughly descending order, here are possible positive results of a change:

- Fix a bug
- Improve performance
- Improve tests
- Clarify user-visible documentation (log/exceptions, javadoc)
- Improve code style or consistency
- Improve code comments or other non-functional clarification

and here are some possible costs or risks:

- introduces a bug
- introduces a performance regression
- adds dependencies
- adds to maintenance overhead
- decreases code consistency
- introduces non-trivial merge conflict problems
- takes time to review and merge

It's not that there aren't positive changes here, in isolation, but they're just marginal. Although interpolation is probably better than format(), it's inconsistent both before and after this change. Same with nonEmpty. Some of these I don't think are improvements, like "fetch" -> "fetches".

But why wouldn't you want to merge any helpful change? It's really the cost of reviewing it, and distantly, the merge conflict issue. I don't mind people getting started by contributing small changes, and paying that cost in order to on-board people into a process and community. I also think the occasional tiny change is fine.

I should also say that I do find it reasonable to fix up code in these small details when already changing it for other purposes.

What I don't want to do is fix 1 issue piecemeal over 10 changes. Either we shouldn't really bother, or should fix it one go as much as possible. For example, although we should probably talk about it, I'd be more in favor of changing all `!...isEmpty` to `nonEmpty` in one go than this change.

I also prefer to see experienced contributors "graduate" towards more significant changes or reviewing/shepherding other changes.

I have heard from a few sources that some organizations may incentivize employees to contribute to Spark (good) and measure by number of JIRAs or pull requests (bad). I don't think that's what's happening here, but another reason I personally want to push back on _everyone_ submitting bitty changes.

**Comment by jaceklaskowski** [comment]
Thanks @rxin and @srowen for your help and patience! I'll close the pull request.

**Pull request closed without merging**